apps/src/code-studio/responsive.js

```diff
@@ -0,0 +1,15 @@
+import debounce from 'lodash/debounce';
+import { getStore } from './redux';
+import { getResponsiveBreakpoint, setResponsiveSize } from './responsiveRedux';
+
+/**
+ * Listen for page resize and dispatch events to Redux when if cross a
+ * responsive breakpoint width.
+ */
+export default function initResponsive() {
+  const store = getStore();
+
+  window.addEventListener('resize', debounce(() => {
+    store.dispatch(setResponsiveSize(getResponsiveBreakpoint(window.innerWidth)));
+  }));
```

Comment:
Two suggestions here:

1. Using the default `wait` value of 0 (only waiting until the next tick) may not improve performance enough for lower-end machines.  I'd suggest a value of at least 50ms (<=20FPS) - especially not knowing how much behavior we'll attach to this event down the road.

2. I might use [`_.throttle()`](https://lodash.com/docs/4.17.4#throttle) instead of debounce here.  With throttle, a very smooth window resize would still trigger intermediate re-renders, albeit at whatever reduced rate we choose.